modules/internal/localeModels/knl/LocaleModel.chpl

```diff
@@ -513,13 +513,13 @@ module LocaleModel {
     proc localeIDtoLocale(id : chpl_localeID_t) {
       const node = chpl_nodeFromLocaleID(id);
       const subloc = chpl_sublocFromLocaleID(id);
-      if (subloc == c_sublocid_none) || (subloc == c_sublocid_any) then
-        return (myLocales[node:int]):locale;
+      if chpl_isActualSublocID(subloc) then
+        return (myLocales[node:int].getChild(subloc:int)):locale;
       else if (subloc == numaDomainForAny(
                         (myLocales[node:int]:LocaleModel).numSublocales)) then
```

Comment:
With the new organization, these two `if` statements need to be inverted.  The `numaDomainForAny()` value passes the `chpl_isActualSublocID()` test, so in the current order, the second `if` statement would never be executed.